main.tf

```diff
@@ -2,10 +2,10 @@ locals {
   create_db_subnet_group = var.db_subnet_group_name == "" ? var.create_db_subnet_group : false
   db_subnet_group_name   = coalesce(var.db_subnet_group_name, module.db_subnet_group.this_db_subnet_group_id)
```

Comment:
@bryantbiggs: It looks like similar change is required for `db_subnet_group_name` as function `coalesce` fails with same error due to 'module.db_subnet_group.this_db_subnet_group_id' parameter which is not yet has a value in case I try to create db_subnet_group.

Reply:
good point, the subnet changes were coming down the pipe as well till this little snafu (https://github.com/terraform-aws-modules/terraform-aws-rds/compare/master...bryantbiggs:fix/subnet-group) - I can put this back to the original though for the time being in this PR

Reply:
updated in https://github.com/terraform-aws-modules/terraform-aws-rds/pull/308/commits/c66fc9cae8c0300eb6ea2cb84d601e6b8882a3d8